internal/global/trace_test.go

```diff
@@ -91,3 +101,101 @@ func TestTraceProviderDelegates(t *testing.T) {
 	gtp.Tracer("abc", trace.WithInstrumentationVersion("xyz"))
 	assert.True(t, called, "expected configured TraceProvider to be called")
 }
+
+func TestTraceProviderDelegates_MultiGoRoutines(t *testing.T) {
+	global.ResetForTest()
+
+	// Retrieve the placeholder TracerProvider.
+	gtp := otel.GetTracerProvider()
+
+	done := make(chan int)
+	quit := make(chan int)
+	go func() {
+		defer close(done)
+		for {
+			select {
+			case <-time.After(1 * time.Millisecond):
+				gtp.Tracer("abc", trace.WithInstrumentationVersion("xyz"))
+			case <-quit:
+				return
+			}
+		}
+	}()
+
+	// Wait for the goroutine to make some calls before installing the provider.
+	<-time.After(100 * time.Millisecond)
```

Comment:
This assumption will be invalid on slow CI systems and will cause buggy tests. Instead having the sending goroutine send a signal or increment a counter that is then watched here for progress will ensure the assumption.

Reply:
Don.e